<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>RSF by dincarnato</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>RSF</h1>
        <p>RNA Structure Framework</p>

        <p class="view"><a href="https://github.com/dincarnato/RSF">View the Project on GitHub <small>dincarnato/RSF</small></a></p>


        <ul>
          <li><a href="https://github.com/dincarnato/RSF/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/dincarnato/RSF/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/dincarnato/RSF">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h1>
<a id="rsf---rna-structure-framework-v120a" class="anchor" href="#rsf---rna-structure-framework-v120a" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>RSF - RNA Structure Framework (v1.2.0a)</h1>

<p>The recent advent of high-throughput methods for probing RNA secondary structures has enabled transcriptome-scale analysis of the RNA structurome.
Despite the establishment of several methods for querying RNA secondary structures on a genome-wide scale (CIRS-seq, SHAPE-seq, Structure-seq, DMS-seq, PARS), no tool has been developed to date to enable the rapid analysis and interpretation of these data.</p>

<p><strong>NEW (version 1.2.0a):</strong>
<strong>RSF now supports the analysis of SHAPE-MaP and RING-MaP data (experimental)</strong></p>

<p>The RNA Structure Framework is a modular toolkit developed to deal with RNA structure probing high-throughput data, from reads mapping to structure inference.<br>
Its main features are: </p>

<ul>
<li>Automatic reference transcriptome creation</li>
<li>Automatic reads preprocessing (adapter clipping and trimming) and mapping</li>
<li>Scoring and data normalization</li>
<li>Accurate RNA folding prediction by incorporating structural probing data</li>
</ul>

<p>For updates, please visit: <a href="https://rsf.hugef-research.org">https://rsf.hugef-research.org</a><br>
For support requests, please post your questions to: <a href="https://groups.google.com/forum/#!forum/rsftoolkit">https://groups.google.com/forum/#!forum/rsftoolkit</a></p>

<h2>
<a id="author" class="anchor" href="#author" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Author</h2>

<p>Danny Incarnato (danny.incarnato[at]hugef-torino.org)<br>
Epigenetics Unit @ HuGeF [Human Genetics Foundation]  </p>

<h2>
<a id="citation" class="anchor" href="#citation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Citation</h2>

<p>Incarnato <em>et al</em>., (2015) RNA structure framework: automated transcriptome-wide reconstruction of RNA secondary structures from high-throughput structure probing data.</p>

<h2>
<a id="license" class="anchor" href="#license" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>License</h2>

<p>This program is free software, and can be redistribute and/or modified under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or any later version.</p>

<p>Please see <a href="http://www.gnu.org/licenses/">http://www.gnu.org/licenses/</a> for more informations.</p>

<h2>
<a id="prerequisites" class="anchor" href="#prerequisites" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Prerequisites</h2>

<ul>
<li>Linux/Mac system</li>
<li>Bowtie v1.0.0 (<a href="http://bowtie-bio.sourceforge.net/index.shtml">http://bowtie-bio.sourceforge.net/index.shtml</a>)</li>
<li>SAMTools v1.2 or greater (<a href="http://www.htslib.org/">http://www.htslib.org/</a>)</li>
<li>BEDTools v2.0 or greater (<a href="https://github.com/arq5x/bedtools2/">https://github.com/arq5x/bedtools2/</a>)</li>
<li>Cutadapt v1.10 or greater (<a href="http://cutadapt.readthedocs.io/en/stable/index.html">http://cutadapt.readthedocs.io/en/stable/index.html</a>)</li>
<li>ViennaRNA Package v2.2.0 or greater (<a href="http://www.tbi.univie.ac.at/RNA/">http://www.tbi.univie.ac.at/RNA/</a>)</li>
<li>RNAstructure v5.6 or greater (<a href="http://rna.urmc.rochester.edu/RNAstructure.html">http://rna.urmc.rochester.edu/RNAstructure.html</a>)</li>
<li>Perl v5.12 (or greater), with ithreads support</li>
<li>
<p>Perl non-CORE modules (<a href="http://search.cpan.org/">http://search.cpan.org/</a>):</p>

<ol>
<li>DBD::MySQL<br>
</li>
<li>LWP::UserAgent<br>
</li>
<li>RNA (part of the ViennaRNA package)<br>
</li>
<li>XML::LibXML<br>
</li>
</ol>
</li>
</ul>

<h2>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Installation</h2>

<p>Clone RSF git repository:</p>

<div class="highlight highlight-source-shell"><pre>git clone https://github.com/dincarnato/RSF</pre></div>

<p>This will create a RSF folder.
To add RSF executables to your PATH, simply type:</p>

<div class="highlight highlight-source-shell"><pre><span class="pl-k">export</span> PATH=<span class="pl-smi">$PATH</span>:/path/to/RSF</pre></div>

<h2>
<a id="usage" class="anchor" href="#usage" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Usage</h2>

<p>Please refer to the RSF Manual.<br>
To obtain parameters list, simply call the required program with the "-h" (or "--help") parameter.</p>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/dincarnato">dincarnato</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
